New:

szymongebala opened the pull request:
-returnBorrowedBooks
Validation
-max quantity of borrowed

szymongebala committed:
New:
-returnBorrowedBooks
Validation
-max quantity of borrowed

bederp commented on pom.xml:
```diff
@@ -67,6 +67,21 @@
 			<version>1.18.12</version>
 			<scope>provided</scope>
 		</dependency>
+        <dependency>
```
3 times junit import?

bederp replied:
Also junit is already provided in spring-boot starter, why are you adding this?

szymongebala replied:
removed

bederp commented on src/main/java/pl/connectis/projektgrupowy/domain/Book.java:
```diff
@@ -7,12 +7,25 @@
 public class Book {
-
+    
+    public Long getId() {
```
Do you need getter if there is @Data annotation?

szymongebala replied:
removed

bederp commented on src/main/java/pl/connectis/projektgrupowy/domain/Client.java:
```diff
@@ -8,21 +8,44 @@
 public class Client {
-
+    
+    public Long getId() {
```
Same as above

szymongebala replied:
removed

bederp commented on src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java:
```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
         Optional<Book> optionalBook = bookRepository.findById(bookId);
-        if (optionalClient.isPresent() && optionalBook.isPresent()) {
+        //added borrow validation
+        if (optionalClient.isPresent() && optionalBook.isPresent()&&optionalClient.get().getBooks().size()<2 ){
```
What if validation is not met, what is propagated back to client?

szymongebala replied:
expection added

bederp commented on src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java:
```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
             clientRepository.save(client);
             return optionalBook.get();
         }
+        System.out.println("No books for u pilgrim");
```
Avoid System.out in production code use Logger

szymongebala replied:
expection added

bederp commented on src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java:
```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
+        
+        
+        
         return null;
```
if client exists return empty collection, if doesn't throw exception and return appropriate http.status to client

szymongebala replied:
added

szymongebala committed:
Removed:
-getters
-System.out.
New:
-exceptions
-admin service
-admin service implementations

szymongebala committed:
Removed:
-getters
-System.out.
New:
-exceptions
-admin service
-admin service implementations

szymongebala committed:
New
-client controller

szymongebala committed:
New
-client controller
